sklearn/linear_model/_coordinate_descent.py

```diff
@@ -792,19 +794,39 @@ def fit(self, X, y, sample_weight=None, check_input=True):
                                      "sparse matrices.")
                 sample_weight = _check_sample_weight(sample_weight, X,
                                                      dtype=X.dtype)
-            # simplify things by rescaling sw to sum up to n_samples
-            # => np.average(x, weights=sw) = np.mean(sw * x)
+            # TLDR: Rescale sw to sum up to n_samples.
+            # Long: The objective function of Enet
+            #
+            #    1/2 * np.average(squared error, weights=sw)
+            #    + alpha * penalty                                   (1)
+            #
+            # is invariant under rescaling of sw.
+            # But enet_path coordinate descent minimizes
+            #
+            #     1/2 * sum(squared error) + alpha * penalty
+            #
+            # and therefore sets
+            #
+            #     alpha = n_samples * alpha
+            #
+            # inside its function body, which results in an objective
+            # equivalent to (1) without sw.
+            # With sw, however, enet_path should set
+            #
+            #     alpha = sum(sw) * alpha                            (2)
+            #
+            # Therefore, using the freedom of Eq. (1) to rescale alpha before
+            # calling enet_path, we do
+            #
+            #     alpha = sum(sw) / n_samples * alpha
+            #
+            # such that the rescaling inside enet_path is exactly Eq. (2)
+            # because now sum(sw) = n_samples.
             sample_weight = sample_weight * (n_samples / np.sum(sample_weight))
-            # Objective function is:
-            # 1/2 * np.average(squared error, weights=sw) + alpha * penalty
-            # but coordinate descent minimizes:
-            # 1/2 * sum(squared error) + alpha * penalty
-            # enet_path therefore sets alpha = n_samples * alpha
-            # With sw, enet_path should set alpha = sum(sw) * alpha
-            # Therefore, we rescale alpha = sum(sw) / n_samples * alpha
-            # Note: As we rescaled sample_weights to sum up to n_samples,
-            #       we don't need this
-            # alpha *= np.sum(sample_weight) / n_samples
+            # Note: Alternatively, we could also have rescaled alpha instead
+            # of sample_weight:
+            #
+            #     alpha *= np.sum(sample_weight) / n_samples
```

Comment:
This reads way better. 👍 Thanks!

Reply:
Thanks likewise, it's based on your suggestion.